### ISSUE TITLE
The concept of Media Display Drivers

### ISSUE THREAD

**Issue by kasvith** [joomla-projects/media-manager-improvement#426]
Currently we display only common file types that are by default rendered by the browser. But there are many file types that are not directly supported by web browsers, So we need to create a Media Display Driver system for developers, So they can create drivers(php and js) to render a particular file type in media  manager. For example someone can create a plugin to display Powerpoint Slide shows, Word documents and so on.

The Media Display Driver can be registered with a file type and rest will be cared by the driver.
We should also allow to edit text files and so on. Everything can be achieved by creating a proper driver system

**Comment by laoneo** [comment]
Would really like to have that, but I'm not sure if we can do that in a way to be extended by plugins. The problem is that it is very connected to vue how media types are rendered. Probably we need to go step by step here and implement #344 first. Then we can see if we can extract the code that way to have it in a Joomla plugin.

**Comment by kasvith** [comment]
yes, or what about Media Manager Specific driver system? I would like to have that for frontend....
We have some interface for frontend driver, we can register vue drivers to it and it will handle the file type associated.

**Comment by dgrammatiko** [comment]
We can use web components for each type, as a way to decouple from vue 😉

**Comment by kasvith** [comment]
sure @dgt41 we need some kind of interface so people can create more drivers.
